--- FunctionCallSignatureSniff.php.orig	2025-10-29 19:18:23.287336481 -0400
+++ FunctionCallSignatureSniff.php	2025-10-29 19:18:02.078007913 -0400
@@ -2,13 +2,18 @@
 /**
  * Ensures function calls are formatted correctly.
  *
+ * Based on PEAR.Functions.FunctionCallSignature, with the following changes:
+ *   - Allows the paren of a multi-line function call to not be the last
+ *     thing on the line if the last thing on the line is an array open
+ *
+ * @author    Alex Howansky <alex.howansky@gmail.com>
  * @author    Greg Sherwood <gsherwood@squiz.net>
  * @copyright 2006-2023 Squiz Pty Ltd (ABN 77 084 670 600)
  * @copyright 2023 PHPCSStandards and contributors
  * @license   https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/licence.txt BSD Licence
  */
 
-namespace PHP_CodeSniffer\Standards\PEAR\Sniffs\Functions;
+namespace PHP_CodeSniffer\Standards\Ork\Sniffs\Functions;
 
 use PHP_CodeSniffer\Files\File;
 use PHP_CodeSniffer\Sniffs\Sniff;
@@ -392,7 +397,11 @@
         }
 
         $next = $phpcsFile->findNext(Tokens::EMPTY_TOKENS, ($openBracket + 1), null, true);
-        if ($tokens[$next]['line'] === $tokens[$openBracket]['line']) {
+        if (
+            $tokens[$next]['line'] === $tokens[$openBracket]['line'] &&
+            $tokens[$next]['code'] !== T_ARRAY &&
+            $tokens[$next]['code'] !== T_OPEN_SHORT_ARRAY
+        ) {
             $error = 'Opening parenthesis of a multi-line function call must be the last content on the line';
             $fix   = $phpcsFile->addFixableError($error, $stackPtr, 'ContentAfterOpenBracket');
             if ($fix === true) {
@@ -405,7 +414,11 @@
 
         $closeBracket = $tokens[$openBracket]['parenthesis_closer'];
         $prev         = $phpcsFile->findPrevious(T_WHITESPACE, ($closeBracket - 1), null, true);
-        if ($tokens[$prev]['line'] === $tokens[$closeBracket]['line']) {
+        if (
+            $tokens[$prev]['line'] === $tokens[$closeBracket]['line'] &&
+            $tokens[$prev]['code'] !== T_ARRAY &&
+            $tokens[$prev]['code'] !== T_CLOSE_SHORT_ARRAY
+        ) {
             $error = 'Closing parenthesis of a multi-line function call must be on a line by itself';
             $fix   = $phpcsFile->addFixableError($error, $closeBracket, 'CloseBracketLine');
             if ($fix === true) {
